#! /bin/sh /usr/share/dpatch/dpatch-run
## 55_top_highlight.dpatch by  <csmall@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix highlighting problem Closes: #351065

@DPATCH@
diff -urNad procps-3.2.7~/top.c procps-3.2.7/top.c
--- procps-3.2.7~/top.c	2008-01-08 20:13:02.000000000 +1100
+++ procps-3.2.7/top.c	2008-01-08 20:13:47.000000000 +1100
@@ -3027,9 +3027,10 @@
       snprintf(_z, sizeof(_z), f, ## va);                                    \
       snprintf(cbuf, sizeof(cbuf), "%s%s%s",                                 \
          q->capclr_rowhigh,                                                  \
-         _z,                                                                 \
+         _z+advance,                                                         \
          !(CHKw(q, Show_HIROWS) && 'R' == p->state) ? q->capclr_rownorm : "" \
       );                                                                     \
+      advance=0;                                                             \
       pad += q->len_rowhigh;                                                 \
       if (!(CHKw(q, Show_HIROWS) && 'R' == p->state)) pad += q->len_rownorm; \
    }                                                                         \
